com.unity.render-pipelines.universal/Documentation~/rendering/deferred-rendering-path.md

```diff
@@ -0,0 +1,422 @@
+    </tr>
+    <tbody>
+</table>
+
+The following sections describe the Deferred Rendering Path render Passes.
+
+#### Depth, or depth and normal prepass
+
+In the Deferred Rendering Path, in the depth prepass or depth and normal prepass, Unity renders only the Materials that do not support the deferred rendering model. For example, Materials using the Complex Lit shader are such Materials.
+
+In the Deferred Rendering Path, Unity does not use the depth prepass to generate a copy of the depth buffer (this behavior is different in the Forward Rendering Path).
+
+If the Universal Renderer has the SSAO Renderer Feature, Unity executes the depth and normal prepass. SSAO uses the screen-space depth and normal buffers to calculate ambient occlusion.
+
+#### Optional passes: SSAO, SSAO with blending
+
+If the Universal Renderer has the SSAO Renderer Feature, and the **After Opaque** option is disabled (the option is disabled by default), Unity executes the SSAO Pass at the AfterRenderingGbuffer event. The SSAO Renderer Feature calculates the SSAO texture. Unity samples this texture in the Deferred rendering Pass and in the Pass that renders Forward-only Materials.
+
+Using this Pass order, Unity can combine the baked occlusion with the real-time occlusion from the SSAO Renderer Feature, and avoid double-darkening from the baked and the real-time ambient occlusion.
+
+When the After Opaque option is enabled, Unity executes the SSAO and blending Pass at the AfterRenderingOpaques event, after rendering the Forward-only Materials. Unity then executes an extra full screen Pass to overlay the SSAO texture onto the Emissive/GI/Lighting buffer. This causes over-darkening of the areas that receive baked occlusion and real-time occlusion.
+
+**Performance considerations**
+
+On mobile platforms with TBDR architecture, with the **After Opaque** option is disabled, Unity requires an extra render target for load and store operations. This has a significant performance impact.
+
+Enabling the **After Opaque** option on mobile platforms improves GPU performance. On mobile platforms with TBDR architecture, enabling this option avoids extra render target load and store operations.
+
+#### Forward-only Pass
+
+Certain Unity shaders use lighting models that Unity cannot render in the Deferred Rendering Path.
+
+Examples of such shaders:
+
+* **Complex Lit**: the Lighting model of this shader (for example, the Clear Coat effect) is too complex and extra Material properties cannot fit into the G-buffer.
+
+* **Baked Lit** and **Unlit**: these shaders do not receive dynamic lighting, so writing them directly into the Lighting buffer instead of the G-buffer is faster.
+
+* **Custom shaders**: Unity renders the shaders that do not declare the Pass tags required by the Deferred Rendering Path as Forward-only. The required Pass tags are: `LightMode`, and `UniversalMaterialType`. For more information, see [URP Pass tags](../urp-shaders/urp-shaderlab-pass-tags.md).
```

Comment:
This information is not very exaplaining, as I user I wouldn't know what to do to create a custom shader that needs to be shaded in forward only.  You should say that that the shader needs a pass with `LightMode` tag should be set to `UniversalForwardOnly`. Then user will be able to know what to look for more details in [URP Pass tags](../urp-shaders/urp-shaderlab-pass-tags.md)

Where do we explain `UniversalMaterialType` and what should be used?

Reply:
#### Forward-only Pass

Certain Unity shaders use lighting models that Unity cannot render in the Deferred Rendering Path.
Unity renders Materials with such shaders in the Forward Rendering Path. These shaders need to declare `UniversalForwardOnly` Lightmode tag so that URP can process them in the forward-only pass instead of the gbuffer and deferred shading passes. For more information, see the section ShaderLab Pass tags.

Examples of such shaders:

* **Complex Lit**: the Lighting model of this shader (for example, the Clear Coat effect) is too complex and extra Material properties cannot fit into the G-buffer.

* **Baked Lit** and **Unlit**: these shaders do not receive dynamic lighting, so writing them directly into the Lighting buffer instead of the G-buffer is faster.

* **Custom shaders**: Unity renders the shaders that do not declare the Pass tags required by the Deferred Rendering Path as Forward-only. The required Pass tags are: `LightMode`, and `UniversalMaterialType`. For more information, see [URP Pass tags](../urp-shaders/urp-shaderlab-pass-tags.md).

For the SSAO Renderer Feature to be able to calculate ambient occlusion for the Materials using the **Complex Lit** shader, Unity must render such Materials in the depth and normal prepass first. This is because Unity does not render those Materials in the G-buffer pass (GBufferPass). For more information, see [URP Pass tags](../urp-shaders/urp-shaderlab-pass-tags.md).

Reply:
> You also mention the shaders that do not declare the Pass tags required by the Deferred Rendering Path as Forward-only but UniversalMaterialType doesn't have correlation to ForwardOnly does it?

@oleks-k  Indeed, UniversalMaterialType has no correlation with ForwardOnly shaders. It should be removed.